Source/Scene/Scene.js

```diff
@@ -1685,64 +1685,137 @@ define([
-                var commands = frustumCommands.commands[pass];
-                var length = frustumCommands.indices[pass];
-                for (var j = 0; j < length; ++j) {
-                    var command = commands[j];
-                    if (command.castShadows) {
-                        shadowCommands.push(command);
+    function getTerrainShadowCommands(scene, shadowMap) {
+        var frameState = scene.frameState;
+
+        var sceneCommandList = frameState.commandList;
+        var sceneCamera = frameState.camera;
+        var sceneCullingVolume = frameState.cullingVolume;
+
+        var terrainCommands = []; // TODO : avoid allocation
+
+        // Update frame state to render from the light camera
+        frameState.commandList = terrainCommands;
+        frameState.camera = shadowMap.shadowMapCamera;
+        frameState.cullingVolume = shadowMap.shadowMapCullingVolume;
+
+        // Collect terrain commands from the light's POV
+        if (scene._globe && scene._globe.castShadows) {
+            scene._globe.update(frameState);
+        }
+
+        // Revert back to original frame state
+        frameState.commandList = sceneCommandList;
+        frameState.camera = sceneCamera;
+        frameState.cullingVolume = sceneCullingVolume;
+
+        return terrainCommands;
+    }
+    
+    // function getTerrainShadowCommands(scene, shadowMap) {
+    //     // TODO : Temporary for testing. Globe.update doesn't work with orthographic frustums currently
+    //     var terrainCommands = [];
+    //     var commandList = scene.frameState.commandList;
+    //     var length = commandList.length;
+    //     for (var i = 0; i < length; ++i) {
+    //         var command = commandList[i];
+    //         if (command.castShadows && command.pass === Pass.GLOBE) {
+    //             terrainCommands.push(command);
+    //         }
+    //     }
+    //
+    //     return terrainCommands;
+    // }
+
+    function insertShadowCommands(commandList, insertAll, shadowMap, shadowPassCommands) {
+        var shadowVolume = shadowMap.shadowMapCullingVolume;
+        var passVolumes = shadowMap.passCullingVolumes;
+
+        var isPointLight = shadowMap.isPointLight;
+        var center = shadowMap.pointLightPosition;
+        var radius = shadowMap.pointLightRadius;
+
+        var numberOfPasses = shadowMap.numberOfPasses;
+
+        var length = commandList.length;
+        for (var i = 0; i < length; ++i) {
+            var command = commandList[i];
+            // Don't insert globe commands with the rest of the scene commands since they are handled separately
+            if (command.castShadows && (insertAll || (command.pass === Pass.OPAQUE || command.pass === Pass.TRANSLUCENT))) {
+                if (isPointLight) {
+                    if (defined(command.boundingVolume)) {
+                        var distance = command.boundingVolume.distanceSquaredTo(center);
+                        if (distance < radius * radius) {
```

Comment:
Store `radius * radius` in a local outside of the loop; I'm not confident that it would get optimized.